woocommerce/woocommerce
# Update notice templates display logic to show updated templates in block themes only

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

I have taken this branch over from @nielslange - thanks Niels for starting the work on this.

This PR will stop the new style of notices showing in classic themes.

In this PR we have made the following changes:

1. In `plugins/woocommerce/src/Blocks/Domain/Services/Notices.php` we now check if `wp_is_block_theme` before registering callbacks for the notice display hooks, that means any classic themes will not have callbacks registered, so the notice templates will load as normal.
2. To facilitate testing of this feature we:
  - created a shortcode checkout and shortcode cart page in the e2e env
  - Added a `storefront-child` and twentytwentyfour-child` them (child themes o one _block_ theme and one _classic_ theme).
  -  Added a rule to `.gitignore` to allow it to be part of the repository.
  - Added `plugins/woocommerce-blocks/tests/e2e/bin/scripts/parallel/themes.sh` to install and activate the themes.
  - Create a coupon during test setup

3. Ensure that template customisations work across block themes and classic themes (i.e. both in the block notice and classic notice)





Closes #44064 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install a block theme (e.g. Twenty Twenty-four) and a classic theme (e.g. Storefront)
2. Add a coupon to your site (any settings are fine).
3. Activate Storefront.
4. Add a page containing the Checkout shortcode and a page containing the Cart shortcode.
5. Add an item to your cart and go to the Cart shortcode page.
6. Apply the coupon. Ensure the notice you see is in the "classic" style

![image](https://github.com/woocommerce/woocommerce/assets/5656702/02e4da47-c245-444c-843a-d4111d4bb744)

7. Try adding the coupon again, ensure the error notice you see is in the "classic" style.

<img width="740" alt="image" src="https://github.com/woocommerce/woocommerce/assets/5656702/1a437be0-ec64-46df-ab30-0ec01d39fa15">

8. Remove the coupon, ensure the notice you see is in the "classic" style.
9. Activate the Twenty Twenty-four theme and repeat the test, however ensure the notice style is the updated style:
<img width="923" alt="image" src="https://github.com/woocommerce/woocommerce/assets/5656702/2ad50364-7b23-4a54-9d7c-4e2eee9b7da6">



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Fixed an issue where the notices shown in classic themes were using the updated styles.

#### Comment 

</details>
